pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -69,6 +69,14 @@ type connectionStore interface {
 	GetConnection(connectionID string) (*ConnectionRecord, error)
 }
 
+// stateMachineMsg is an internal struct used to pass data to state machine.
+type stateMachineMsg struct {
```

Comment:
are we planning to keep a combined inbound/outbound structure within the state machine?

Reply:
We can separate [Execute](https://github.com/hyperledger/aries-framework-go/blob/master/pkg/didcomm/protocol/didexchange/states.go#L54) function into ExecuteInbound and ExecuteOutbound.

Reply:
Created an issue - https://github.com/hyperledger/aries-framework-go/issues/501